linux/scripts/dist.sh

```diff
@@ -69,54 +69,50 @@ for proj in ${autotool_projects}; do
 done
 
 for proj in ${extra_projects}; do
-    # dist for keyman-config
-    if [ "${proj}" == "keyman-config" ]; then
-        cd keyman-config
-        echo "3.0 (native)" > debian/source/format
-        dch keyman-config --newversion ${VERSION} --force-bad-version --nomultimaint
-        dpkg-source --tar-ignore=*~ --tar-ignore=.git --tar-ignore=.gitattributes \
-            --tar-ignore=.gitignore --tar-ignore=experiments --tar-ignore=debian \
-            --tar-ignore=buildtools/build-langtags.py --tar-ignore=__pycache__ -Zgzip -b .
-        mv ../keyman-config_*.tar.gz ../dist/keyman-config-${VERSION}.tar.gz
-        echo "3.0 (quilt)" > debian/source/format
-    elif [ "${proj}" == "keyboardprocessor" ]; then
-        cd ../common/core
-        kbpvers="keyman-keyboardprocessor-$VERSION"
-        cp -a desktop $kbpvers
-        cp ../../VERSION.md $kbpvers
-        cp ../../TIER.md $kbpvers
-        mkdir -p $kbpvers/scripts
-        cp ../../resources/shellHelperFunctions.sh $kbpvers/scripts
-        tar cvzf $kbpvers.tar.gz --exclude=debian --exclude=build --exclude=.gitignore $kbpvers
-        rm -rf $kbpvers
-        cp $kbpvers.tar.gz ../../linux/dist
-    fi
+    # dist for keyman
+    cp -a debian ../
+    cd ..
+    echo "3.0 (native)" > debian/source/format
+    dch keyman --newversion ${VERSION} --force-bad-version --nomultimaint
+    dpkg-source --tar-ignore=*~ --tar-ignore=.git --tar-ignore=.gitattributes \
+        --tar-ignore=.gitignore --tar-ignore=experiments --tar-ignore=debian \
+        --tar-ignore=.github --tar-ignore=.vscode --tar-ignore=android \
+        --tar-ignore=common/core/web --tar-ignore=common/lexical-model-types \
+        --tar-ignore=common/models --tar-ignore=common/predictive-text \
+        --tar-ignore=common/test --tar-ignore=developer --tar-ignore=docs --tar-ignore=ios \
+        --tar-ignore=linux/keyman-config/buildtools/build-langtags.py --tar-ignore=__pycache__ \
+        --tar-ignore=linux/help --tar-ignore=linux/ibus-kmfl --tar-ignore=linux/Jenkinsfile \
+        --tar-ignore=linux/keyboardprocessor --tar-ignore=linux/kmflcomp \
+        --tar-ignore=linux/libkmfl --tar-ignore=mac --tar-ignore=node_modules --tar-ignore=oem \
+        --tar-ignore=linux/build* --tar-ignore=common/core/desktop/build \
+        --tar-ignore=resources/devbox --tar-ignore=resources/git-hooks \
+        --tar-ignore=resources/scopes --tar-ignore=resources/web-environment \
+        --tar-ignore=resources/build/*.lua --tar-ignore=resources/build/jq* \
+        --tar-ignore=resources/build/vswhere* --tar-ignore=results \
+        --tar-ignore=web --tar-ignore=windows --tar-ignore=keyman_1* \
+        --tar-ignore=dist --tar-ignore=.pbuilderrc --tar-ignore=VERSION \
+        --tar-ignore=scripts -Zgzip -b .
```

Comment:
This is pretty fragile, because as we add other files and folders in the root we have to remember to update this, buried away in /linux. Is there a way we can improve this? For example, making a temp folder which has a copy of specific files and folders to _include_, and working from that? That's perhaps not entirely wonderful either but I think this as it stands is going to need constant maintenance for unrelated work.

Reply:
It doesn't really matter if there are additional files and folders in the source package - just makes it a little bigger. But if we copy specific files and put those in the source package we might miss added necessary files which would cause build failures. This needs maintenance in both cases, but in the former we can do the maintenance when we get to it. I think I'd prefer to leave it as is.

Reply:
ok fair enough